test/useAsObservableSource.test.tsx

```diff
@@ -356,3 +356,30 @@ it.skip("checks for stable shape of object being passed in", async () => {
     )
     restore()
 })
+
+describe("enforcing actions", () => {
```

Comment:
I think it's unnecessary to test `useAsObservableSource` in this case since it's used `useLocalStore` internally. It shouldn't break coverage.

Reply:
I found that both `useAsObservableSource` and`useLocalStore` modify observables directly in different ways. Since someone could call `useAsObservableSource` directly, I think it makes sense to test both hooks independently. It is also possible that `useAsObservableSource` could be removed from `useLocalStore` in the future (however unlikely) reducing coverage and introducing a regression.

I took another look to see if I could optimize the `runInAction` call out (internal to `useAsObservableSourceInternal`) if `usedByLocalStore` was true, but since `useLocalStore` needs to call `useAsObservableSource` at the top level I don't think I can since I'd need to wrap it in `runInAction` to make it work ☹️